rootex/framework/components/hierarchy_component.cpp

```diff
@@ -47,6 +47,25 @@ bool HierarchyComponent::addChild(Ref<Entity> child)
 	return false;
 }
 
+bool HierarchyComponent::setupEntities()
+{
+	if (m_Owner->getID() != ROOT_ENTITY_ID)
+	{
+		m_Parent = EntityFactory::GetSingleton()->findEntity(m_ParentID)->getComponent<HierarchyComponent>().get();
```

Comment:
Something like:

```cpp
bool status = true;

// here comes the function you have written which changes status to false in case nullptr is found

return status;
```